pkgs/development/compilers/chicken/5/default.nix

```diff
@@ -1,19 +1,39 @@
-{ lib, newScope } :
+{ lib, newScope, fetchzip }:
 let
   callPackage = newScope self;
 
-  self = {
+  self = with lib; {
     pkgs = self;
 
-    fetchegg = callPackage ./fetchegg { };
+    fetchegg = { name, version, sha256, ... }:
+      fetchzip {
```

Comment:
I prefer to use `fetchurl` and pass `hash` instead of `sha256` here.

Reply:
`fetchurl` does not unzip the file and we need it unzipped for `eggDerivation`. But we could switch to `hash` if `sha256` is a dealbreaker.

Reply:
> `fetchurl` does not unzip the file and we need it unzipped for `eggDerivation`.

Which means that `eggDerivation` doesn't inherit the default `unpackPhase`?

> But we could switch to `hash` if `sha256` is a dealbreaker.

Definitely not.

Reply:
> Which means that eggDerivation doesn't inherit the default unpackPhase?

It uses `stdenv.mkDerivation`. I will try plain `fetchurl` out later today.

Reply:
That worked well. It not uses `fetchurl`.